pkg/hive/manager.go

```diff
@@ -5,50 +5,42 @@ package hive
 
 import (
 	"context"
+	"encoding/json"
 
-	"github.com/gofrs/uuid"
 	hiveclient "github.com/openshift/hive/pkg/client/clientset/versioned"
+	icazure "github.com/openshift/installer/pkg/asset/installconfig/azure"
 	"github.com/sirupsen/logrus"
-	corev1 "k8s.io/api/core/v1"
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kruntime "k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/util/retry"
 
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
 	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
 )
 
 type ClusterManager interface {
-	CreateNamespace(ctx context.Context) (*corev1.Namespace, error)
-	CreateOrUpdate(ctx context.Context, parameters *CreateOrUpdateParameters) error
-	Delete(ctx context.Context, namespace string) error
-}
-
-// CreateOrUpdateParameters represents all data in hive pertaining to a single ARO cluster.
-// CreateOrUpdate must not receive any data which requires an API call to the customer cluster
-// as the intention of this is to be able to reconcile hive resources from CosmosDB -> Hive
-// and this process should work even if the customer cluster is not responding for any reason.
-type CreateOrUpdateParameters struct {
-	Namespace                  string
-	ClusterName                string
-	Location                   string
-	InfraID                    string
-	ClusterID                  string
-	KubeConfig                 string
-	ServicePrincipal           string
-	APIServerPrivateEndpointIP string
+	CreateOrUpdate(context.Context, *api.OpenShiftCluster, *api.Subscription) error
```

Comment:
I added some comments about why they don't go in the constructor in here -- when you patch the document, the changes don't show up, since something somewhere doesn't do the reassign. So, the most reliable thing for installing is to pass them in here.